### ISSUE TITLE
No Longer Working

### ISSUE THREAD

**Issue by L0rdC0rx** [unintended/download-organizer-chrome-extension#40]
In the last couple of weeks this addin has completely stopped working for me.
All I have 1 rule, based on the referrer URL that places it in a subfolder of the Downloads, based on a number in this referrer url. It was all working fine till about 1 or 2 weeks ago, and now it does nothing, not even a hard coded value works. Perhaps the referrer URL pickup is broken?

**Comment by Gapkobob** [comment]
Trying to set up https?://docs\.google\.com/.* , doesnt work as well...

**Comment by L0rdC0rx** [comment]
The whole addin looks like it no longer works since likely some change in Chrome.....
Disappointed, as this is VERY useful when it works, and I do rely on it quite a bit to make my life easier!

**Comment by unintended** [comment]
You can debug the addon and check what happens:
1. Open chrome://extensions/
1. Open inspect view and check console ![image](https://user-images.githubusercontent.com/641973/97117295-beedb900-1713-11eb-8e79-541f2d89659b.png)
1. Download something and you will get your log ![image](https://user-images.githubusercontent.com/641973/97117265-95349200-1713-11eb-8eea-6fd394632e92.png)

**Comment by L0rdC0rx** [comment]
Thanks, I did this, and it seems the referrer in the URL for the site I use this, is NOT the full URL of the site I am on. I use this with a ticket system, so download go to their own folder per case number, and the referrer URL must have been the full url (with the case number) and now, it's just the base url... not sure if this is an issue with Chrome or something else...

**Comment by L0rdC0rx** [comment]
Doing some tests, the referrer url is only ever the base URL and never includes any more of the path.... I checked this on a variety of sites....

**Comment by L0rdC0rx** [comment]
Screenshot of what I see:

![2020-10-26 10-55-11 213](https://user-images.githubusercontent.com/1747417/97164377-f1320180-1779-11eb-8d49-8908a9ea403f.png)

**Comment by unintended** [comment]
yep, looks like it's not possible now - the extension doesn't have enough data to sort it out (I don't know if it's chrome or zendesk got broken)

**Comment by L0rdC0rx** [comment]
my guess is Chrome, as I see the same - only seeing the base URL on EVERY site I try, not just zendesk.
Is there any setting in Chrome that might control this, or anything you can do to get the full URL of the page you are on?

**Comment by unintended** [comment]
Version 81.0.4044.113 (Official Build) (64-bit) works well at github code downloads
![image](https://user-images.githubusercontent.com/641973/97184852-6c160f00-17b0-11eb-811f-f85210ee585b.png)

**Comment by unintended** [comment]
Version 86.0.4240.111 (Official Build) (x86_64) is still ok, so I think it depends on the site and download mechanic (maybe if download is initiated via js it can break)
![image](https://user-images.githubusercontent.com/641973/97185280-eb0b4780-17b0-11eb-9de5-ee7d54d310fa.png)

**Comment by L0rdC0rx** [comment]
I found the problem
it's due to the new Chrome Referrer behaviour - https://developers.google.com/web/updates/2020/07/referrer-policy-new-chrome-default when I disabled this flag, then it worked....

**Comment by L0rdC0rx** [comment]
is there maybe something you can do to add in an additional field for the rule, to detect the open tab, rather than rely on the referrer in the downloaditem object?

**Comment by unintended** [comment]
I'm not sure it's technically possible to do this within an extension, the only and the proper way to do is to monitor and match fields in the downloaditem object. I will not add any extra logic (maybe I will add info about the flag into help section)

**Comment by L0rdC0rx** [comment]
ok, for now, as this addin is very useful, I did disable this flag, and it works now for me....
